src/Signature/SignatureV4.php

```diff
@@ -82,6 +82,7 @@ public function presign(
         $expires,
         array $options = []
     ) {
+        $expiresTimestamp = $this->convertToTimestamp($expires);
```

Comment:
@kstich Not sure I follow what the fix for both scenarios would be.  I think part of the issue is that if the SDK allows any relative timestamp to be passed there will always be scenarios when the duration for the expires length could be off.  The reason for that is that the thread could be paused/roll over a second at any given point, so unless the caller of presign includes a "now" parameter (options perhaps?) there is always the possibility of a slight skew.

We could add that as an option to pass, if passed any strtotime in the convertToTimestamp will use the "now" as the second parameter for the time it is based on.  

This would solve any specific duration requests from possibly being off by a tiny amount.  We could maintain BC by making it an optional parameter, so if it isn't passed it will just use the current time that the createSign is invoked, which majority of the time would be the same time.  

Let me know if that makes sense.  I think the change to the existing PR would be fairly minimal.  I'd update convertToTimestamp to take a second parameter, which would be passed to the strtotime conversion in it.  Then in the options array it'd check for the "now" parameter, if non-existent it'd take current time.  What do you think the option should be?  "relativeTimeBase" could be more descriptive than now?